include/aikido/planner/vectorfield/MoveEndEffectorOffsetVectorField.hpp

```diff
@@ -12,70 +12,58 @@ namespace vectorfield {
 /// This class defines two callback functions for vectorfield planner.
 /// One for generating joint velocity in MetaSkeleton state space,
 /// and one for determining vectorfield planner status.
-class MoveEndEffectorOffsetVectorField
+class MoveEndEffectorOffsetVectorField : public ConfigurationSpaceVectorField
 {
 public:
   /// Constructor
   ///
+  /// \param[in] _stateSpace MetaSkeleton state space
   /// \param[in] _bn Body node of end-effector
-  /// \param[in] _linearVelocity Linear velocity in all the directions
-  /// \param[in] _startTime Start time of a planned trajectory
-  /// \param[in] _endTime End time of a planned trajectory
-  /// \param[in] _timestep Time step size
-  /// \param[in] _linearGain Gain of P control on linear deviation
-  /// \param[in] _linearTolerance Tolerance on linear deviation
-  /// \param[in] _angularGain Gain of P control on angular deviation
-  /// \param[in] _angularTolerance Tolerance on angular deviation
+  /// \param[in] _direction Unit vector in the direction of motion
+  /// \param[in] _distance Minimum distance in meters
+  /// \param[in] _maxDistance Maximum distance in meters
+  /// \param[in] _positionTolerance Constraint tolerance in meters
+  /// \param[in] _angularTolerance Constraint tolerance in radians
+  /// \param[in] _linearVelocityGain Linear velocity gain in workspace.
+  /// \param[in] _initialStepSize Initial step size
+  /// \param[in] _jointLimitTolerance If less then this distance to joint
+  /// limit, velocity is bounded in that direction to 0
   /// \param[in] _optimizationTolerance Tolerance on optimization
-  /// \param[in] _padding Padding to the boundary
   MoveEndEffectorOffsetVectorField(
+      aikido::statespace::dart::MetaSkeletonStateSpacePtr _stateSpace,
       dart::dynamics::BodyNodePtr _bn,
-      const Eigen::Vector3d& _linearVelocity,
-      double _startTime,
-      double _endTime,
-      double _timestep,
-      double _linearGain = 1.,
-      double _linearTolerance = 0.01,
-      double _angularGain = 1.,
-      double _angularTolerance = 0.01,
-      double _optimizationTolerance = 1e-3,
-      double _padding = 1e-5);
+      const Eigen::Vector3d& _direction,
+      double _distance,
```

Comment:
This is following the naming in `prpy`. This is usually the expected distance to move. The planner will terminate once this distance is reached. It will fail it `_maxDistance` is reached.